v2/util/qc_stmt.go

```diff
@@ -117,6 +119,76 @@ func checkAsn1Reencoding(i interface{}, originalEncoding []byte, appendIfCompari
 	return result
 }
 
+func GetEtsiQcTypes(c *x509.Certificate) []asn1.ObjectIdentifier {
+	var result []asn1.ObjectIdentifier
+	ext := GetExtFromCert(c, QcStateOid)
+	if ext == nil {
+		return nil
+	}
+	s := ParseQcStatem(ext.Value, IdEtsiQcsQcType)
+	if len(s.GetErrorInfo()) != 0 {
+		return nil
+	}
+	if !s.IsPresent() {
+		return result
+	}
+	qcType := s.(Etsi423QcType)
+	result = append(result, qcType.TypeOids...)
+	return result
+}
+
+func HasCertAnyEtsiQcStatement(c *x509.Certificate) bool {
+	ext := GetExtFromCert(c, QcStateOid)
+	if ext == nil {
+		return false
+	}
+	return IsAnyEtsiQcStatementPresent(ext.Value)
+}
```

Comment:
If/when IsAnyEtsiQcStatementPresent is updated (e.g. to take an x509.Certificate) this function wouldn't be needed.

It could just do a case over c.QCStatements.StatementIDs